lib/runtime.cpp

```diff
@@ -208,7 +217,45 @@ RuntimeSettings::RuntimeSettings(const std::string& prefix_, const std::string&
         logging_config_file = assert_file(default_logging_config_file, "Default logging config");
     }
 
-    www_dir = prefix / defaults::DATAROOT_DIR / defaults::NAMESPACE / defaults::WWW_DIR;
+    const auto settings_controller_port_it = settings.find("controller_port");
+    if (settings_controller_port_it != settings.end()) {
+        controller_port = (*settings_controller_port_it).get<std::string>();
+    } else {
+        controller_port = defaults::CONTROLLER_PORT;
+    }
+
+    const auto settings_mqtt_server_it = settings.find("mqtt_server");
+    if (settings_mqtt_server_it != settings.end()) {
+        mqtt_server = (*settings_mqtt_server_it).get<std::string>();
+    } else {
+        mqtt_server = defaults::MQTT_SERVER;
+    }
+
+    const auto settings_mqtt_port_it = settings.find("mqtt_port");
+    if (settings_mqtt_port_it != settings.end()) {
+        mqtt_port = (*settings_mqtt_port_it).get<std::string>();
+    } else {
+        mqtt_port = defaults::MQTT_PORT;
+    }
+
+    const auto settings_mqtt_everest_prefix_it = settings.find("mqtt_everest_prefix");
+    if (settings_mqtt_everest_prefix_it != settings.end()) {
+        mqtt_everest_prefix = (*settings_mqtt_everest_prefix_it).get<std::string>();
```

Comment:
why not settings_mqtt_everest_prefix_it->get ?

Reply:
why not indeed